src/raw_pixels.rs

```diff
@@ -71,12 +71,38 @@ impl Iterator for RawPixels<'_, '_> {
         match self.raw_bmp.color_bpp() {
             Bpp::Bits1 => self.pixel_data.get(byte_idx).map(|byte| {
                 let mask = 0b_1000_0000 >> self.bit_idx % 8;
-                pixel_value[0] = (byte & mask != 0) as u8;
+
+                let bit = byte & mask != 0;
+
+                if let Some(table) = self.raw_bmp.header().color_table {
+                    // Color mapping - look into table for 0/1 mapped color
+                    let bit = bit as usize;
+
+                    // Each color table entry is 4 bytes long
+                    let offset = bit * 4;
+
+                    pixel_value[0] = table[offset + 0];
+                    pixel_value[1] = table[offset + 1];
+                    pixel_value[2] = table[offset + 2];
+                } else {
+                    // No color mapping - use on/off value directly
+                    pixel_value[0] = bit as u8;
+                }
+            }),
+            Bpp::Bits8 => self.pixel_data.get(byte_idx).map(|byte| {
+                // Color mapping - look into table for mapped color
+                if let Some(table) = self.raw_bmp.header().color_table {
+                    // Each color table entry is 4 bytes long
+                    let offset = *byte as usize * 4;
+
+                    pixel_value[0] = table[offset + 0];
+                    pixel_value[1] = table[offset + 1];
+                    pixel_value[2] = table[offset + 2];
+                } else {
+                    // No color mapping - use value directly
+                    pixel_value[0] = *byte
+                }
```

Comment:
I'm sure there's a better way to do these indexing operations, but I'm pretty tired at the moment.

Reply:
As mentioned before, I would not mix the color table lookup and the `RawPixels` iter. Instead `RawPixels` would always return the raw image data, which would be the index into the color table for images with a pallete. The iterator could then be mapped to the actual color values by looking up the index in the color table. This step would be optional and shouldn't be used for true color images to improve performance.

Reply:
That makes sense. Are you expecting the colour table logic to go into the `Pixels` iterator impl? I'm unsure how we'd be able to make the iterator zero cost for images without a colour table. Did you have a particular impl in mind?

Reply:
The color table logic would need to go into the `Pixels` iterator impl and it couldn't be zero cost in this instance.

But we don't need to use the `Pixels` iterator in the `ImageDrawable` implementation. `RawBmp::draw` could use `RawPixels` instead of `Pixels` and have different code paths for image with or without a color table, like this pseudo code:
```rust
    pub(crate) fn draw<D>(&self, target: &mut D) -> Result<(), D::Error>
    where
        D: DrawTarget,
        D::Color: From<<D::Color as PixelColor>::Raw>,
    {
        if has_color_table {
            target.fill_contiguous(
                &Rectangle::new(Point::zero(), self.size()),
                self.pixels().map(|Pixel(_, color)| do_color_mapping(color)),
            )
        } else {
            target.fill_contiguous(
                &Rectangle::new(Point::zero(), self.size()),
                self.pixels().map(|Pixel(_, color)| D::Color::Raw::from_u32(color).into()),
            )
        }
    }
```